target/snitch_cluster/sw/apps/snax-gemm-engine/src/snax-gemm-engine.c

```diff
@@ -0,0 +1,69 @@
+
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "data.h"
+#include "snrt.h"
+
+int main() {
+    // Set err value for checking
+    int err = 0;
+
+    // Prepare addresses in TCDM
+    uint8_t *local_a, *local_b;
+    uint32_t *local_c;
+
+    uint32_t tic, toc;
+
+    // Allocate space in TCDM
+    local_a = (uint8_t *)snrt_l1_next();
+    local_b = local_a + m * k * sizeof(uint8_t);
+    local_c = (uint32_t *)(local_b + n * k * sizeof(uint8_t));
```

Comment:
Why would the writeback of let's say **7** result values, not be a problem here, but it would be a problem in the HWPE mac engine case?

Reply:
Good question @JosseVanDelm!

As far as I know @xiaoling-yi made sure that the controller packs the data back correctly, filling up the entire 64-bit access first before sending it back to the memory.

On 2nd thought....

@xiaoling-yi the assumption for your accelerator is that it needs to pack 8 data right? Because your memory reads is always fixed for the assumption that GEMM data is 8 bits and we always need to make sure we get 8, 8-bit data for every line access in the data memory right?

Reply:
Yes, you are right @rgantonio. The data and corresponding address are arranged already by GEMM so we can only assign a start pointer.

Reply:
Yes @JosseVanDelm , for GEMM, the output is 64 * 32, so it is already 64 aligned.